### ISSUE TITLE
time-consuming of the FID computation

### ISSUE THREAD

**Issue by LonglongaaaGo** [mseitzer/pytorch-fid#67]
time-consuming of the FID computation
Hi, I want to know why the FID computation is very slow. When I calculate the FID of 28000 images, it sometimes got stuck and spent almost one day or more to calculate once! Is there any idea to help me fix this problem? Thanks!

**Comment by LonglongaaaGo** [comment]
BTW, I want to know if fake data numbers and real data numbers are not equal, will it affect the calculation of FID?

**Comment by mseitzer** [comment]
It is hard to debug this from afar because it can depend on many different factors. In general, I don't see an intrinsic reason why it should get stuck at some points (computation should be equally fast over time). You can install [tqdm](https://github.com/tqdm/tqdm) to get a better overview how the computation progresses. You could also try to profile the code using [cProfile](https://docs.python.org/3/library/profile.html).

**Comment by LonglongaaaGo** [comment]
Sorry for reply late. 
I found when I am running the codes. It will get stuck in the "calculate_frechet_distance" can you give me some insights to solve this problem?
Thanks again for your consideration!

**Comment by LonglongaaaGo** [comment]
I am running these codes in the SLURM cluster.

**Comment by Totoro97** [comment]
@LonglongaaaGo Hi, I faced the similar problem and I found that the program got stuck in scipy.linalg.sqrtm. Reinstalling Scipy with newest version works for me.

**Comment by Cerf-Volant425** [comment]
I also met the same problem ...

**Comment by Cerf-Volant425** [comment]
oh then I solved it by changing a virtual environment ...

**Comment by chensjtu** [comment]
> 

I wonder the right version of scipy

**Comment by yuxu915** [comment]
> > I wonder the right version of scipy
> 
> spicy 1.5.4 worked for me.

also worked for me. Thanks!

**Comment by aykborstelmann** [comment]
If anyone else already still has the problem after installing the newest scipy version: 
In https://github.com/scipy/scipy/issues/14594#issuecomment-906145042 the issue was found by setting the environment variable `MKL_NUM_THREADS=1` using `export MKL_NUM_THREADS=1`

**Comment by z-fabian** [comment]
I had issues even with newer `scipy` version and setting `MKL_NUM_THREADS=1`. What eventually worked for me is installing `numpy==1.23.3`.  Currently, I have `scipy==1.9.1` and `numpy==1.23.3` and it is not getting stuck in `scipy.linalg.sqrtm` anymore.

**Comment by j-cyoung** [comment]
I met the same issue. I have a server with 4 gpus, and I ran 4 programs on each gpu on parallel. Each program will compute FID. I found that all cpus will be occupied when running multi process with FID computing in the same time, but one process ran well. I think there might be some threads conflicts between torch, scipy.linalg.sqrtm (in FID) and multi-process/threads.

Setting "export MKL_NUM_THREADS=1" didn't help. But I found that setting "export OPENBLAS_NUM_THREADS=1" or "export OMP_NUM_THREADS=1" does help for me. So anyone who has the same issue can give a try.

Related Issue: https://github.com/numpy/numpy/issues/8120#issuecomment-252058015
Related Blog: https://pythonspeed.com/articles/concurrency-control/
